server/handlers/prospect.js

```diff
@@ -1,18 +1,26 @@
+import _ from "lodash";
+import Promise from "bluebird";
+
 import Clearbit from "../clearbit";
 
 export default function handleProspect({ hostSecret }) {
   return (req, res) => {
-    const { domains, role, seniority, titles } = req.body;
+    const { domains, role, seniority, titles = [], limit } = req.body;
     const { client: hull, ship } = req.hull;
-
+    let newLimit = limit;
     if (domains) {
       const cb = new Clearbit({ hull, ship, hostSecret });
-      const prospecting = domains.map(domain => {
-        return cb.fetchProspects({ domain, role, seniority, titles });
+      const prospecting = Promise.mapSeries(domains, (domain) => {
+        const params = { domain, role, seniority, titles, limit: newLimit };
+        return cb.fetchProspects(params).then((ret) => {
+          newLimit -= ret.length;
+          return ret;
+        });
       });
-      Promise.all(prospecting).then(prospects =>
-        res.json({ prospects })
-      ).catch((error) => {
+
+      Promise.all(prospecting).then((results) => {
+        res.json({ prospects: _.flatten(results) });
```

Comment:
I think we can skip those `{` `}` brackets since we are doing only one operation